src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -168,4 +168,22 @@ public void WritePortBindingEnvironmentVariables(IResource resource)
             }
         }
     }
+
+    internal void WriteManifestMetadata(IResource resource)
+    {
+        if (!resource.TryGetAnnotationsOfType<ManifestMetadataAnnotation>(out var metadataAnnotations))
+        {
+            return;
+        }
+
+        Writer.WriteStartObject("metadata");
+
+        foreach (var metadataAnnotation in metadataAnnotations)
+        {
+            Writer.WritePropertyName(metadataAnnotation.Name);
+            JsonSerializer.Serialize(Writer, metadataAnnotation.Value);
```

Comment:
RIP AOT 😄

Reply:
We can probably inject a serializer via DI here but TBH it don't think anyone would be doing AOT on their app host, given the product we are shipping for GA there is no reason to wait for the native build when you just want to launch the app to either debug locally or produce a manifest.